### ISSUE TITLE
Return 5xx error types in HttpResults

### ISSUE THREAD

**Issue by ryancormack** [aws/aws-lambda-dotnet#1431]
### Describe the bug

Currently the HttpResults class doesn't have any status codes for returning 5xx (probably specifically 500 in my example case). I was unsure if this is a specific design decision or not. 

It seems like the IHttpResult is still [really new](https://github.com/aws/aws-lambda-dotnet/issues/979#issuecomment-1423162659).

I tend to wrap my handling code in a try catch block to prevent uncaught exceptions being bubbled up and giving better control over error messages to various clients. Without being able to return an `HttpResults.InternalServerError("some error")`, it limits the ability to use the `IHttpResult` return type. E.g.:

```csharp
[LambdaFunction]
    [RestApi(LambdaHttpMethod.Post, "/command")]
    public async Task<IHttpResult> FunctionHandlerAsync([FromBody] string data, ILambdaContext context)
    {
        try
        {
            var result = await _handler.Handle(data);

            return HttpResults.Ok(JsonSerializer.Serialize(result.ResponseMessage));
        }
        catch (Exception e)
        {
            _logger.Error(e, "My log detail");
            return HttpResults.InternalServerError("Potential friendly error); //No such method
            return new APIGatewayProxyResponse //Not assignable to IHttpResult
            {
                Body = "Potential friendly error",
                StatusCode = (int)HttpStatusCode.InternalServerError,
            };
        }
```

### Expected Behavior

The ability to return a `HttpResults.InternalServerError`

### Current Behavior

Method does not exist

### Reproduction Steps

Example code above

### Possible Solution

Adding various 5xx types to the HttpResults class

### AWS .NET SDK and/or Package version used

Amazon.Lambda.Annotations 0.11.0

### Targeted .NET Platform

Net Standard 2.0/NET6

### Operating System and version

All

**Comment by ashishdhingra** [comment]
The associated PR https://github.com/aws/aws-lambda-dotnet/pull/1432 appears to add few more HTTP status codes in addition to what were added in released PR https://github.com/aws/aws-lambda-dotnet/pull/1421. Needs review with the team.

**Comment by normj** [comment]
The PR has been released as part of version 0.12.0 of Amazon.Lambda.Annotations.
